test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -480,3 +480,237 @@ def test_res_size_runtime(self):
         job_comment += " foobar (R: 95gb A: 50gb T: 50gb)"
         a = {'job_state': 'Q', 'comment': job_comment}
         self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
+
+    def test_res_size_invalid_input(self):
+        """
+        Test invalid values returned from server_dyn_resource
+        script for resource type 'size'.
+        Script returns a 'string' instead of type 'size'.
+        """
+        # Create a resource of type size
+        resname = ["foobar"]
+        restype = ["size"]
+        resflag = ["q"]
+
+        # Script returns invalid value for resource type 'size'
+        script_body = "echo two gb"
+
+        fn = self.du.create_temp_file(prefix="PtlPbs_size",
+                                      suffix=".scr",
+                                      body=script_body)
+        self.du.chmod(path=fn, mode=0755, sudo=True)
+
+        resval = ['"' + resname[0] + ' ' + '!' + fn + '"']
+
+        self.setup_dyn_res(resname, restype, resval, resflag)
+
+        # Submit job
+        a = {'Resource_List.foobar': '2gb'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        job_comment = "Can Never Run: Insufficient amount of server resource:"
+        job_comment += " foobar (R: 2gb A: 0kb T: 0kb)"
+        a = {'job_state': 'Q', 'comment': job_comment}
+        self.server.expect(JOB, a, id=jid, attrop=PTL_AND)
```

Comment:
You're probably doing the right test here, but I'll offer something else to look for as well.  There is an error log message printed at query time when a server_dyn_res script returns an invalid value ("Script <script> returned bad output").

Reply:
Added this check